internal/adapters/gitlabtokenrefresher/gitlabtokenrefresher.go

```diff
@@ -0,0 +1,105 @@
+package gitlabtokenrefresher
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/SwissDataScienceCenter/renku-gateway-v2/internal/adapters/redisadapters"
+	"github.com/SwissDataScienceCenter/renku-gateway-v2/internal/models"
+	"github.com/go-co-op/gocron"
+)
+
+type tokenResponse struct {
+	AccessToken  string `json:"access_token"`
+	Type         string `json:"token_type"`
+	ExpiresIn    int64  `json:"expires_in"`
+	RefreshToken string `json:"refresh_token"`
+	Scope        string `json:"scope"`
+	CreatedAt    int64  `json:"created_at"`
+}
+
+func ScheduleRefreshExpiringTokens(ctx context.Context, redisAdapter redisadapters.RedisAdapter, gitlabTokenRefreshURL string, clientID string, clientSecret string, minsToExpiration int) error {
```

Comment:
See my other comment use an interface, not the concrete type for the `RedisAdapter`.